### PR TITLE
Perform evaluation of src coefficients

### PR DESCRIPTION
**Pull request by sofianehaddad**
Now, we can compute :
 - SRC coefficients using the standard formula
 - SRC with scale :  coefficients sum to 1
 - signed SRC coefficients taking into account sign of linear expansion.

This fixes http://trac.openturns.org/ticket/980

### CODE REVIEW

File: python/src/CorrelationAnalysis_doc.i.in [hunk 1]
```diff
@@ -109,6 +109,10 @@ sample1 : 2-d sequence of float
     First sample, of dimension :math:`n \geq 1`.
 sample2 : 2-d sequence of float
     Second sample, of dimension = 1.
+scale : bool
+    if True, coefficients sums to one.
```
Comment (jschueller) [inline python/src/CorrelationAnalysis_doc.i.in]: coefficients sum to one

File: python/src/CorrelationAnalysis_doc.i.in [hunk 2]
```diff
@@ -133,6 +137,10 @@ sample1 : 2-d sequence of float
...
```
Comment (jschueller) [inline python/src/CorrelationAnalysis_doc.i.in]: here too

File: python/src/CorrelationAnalysis_doc.i.in (hunk 1 above)
Comment (sofianehaddad) [reply python/src/CorrelationAnalysis_doc.i.in]: indeed

File: ChangeLog [hunk 3]
```diff
@@ -23,6 +23,7 @@
  * #979 (Viewer does not handle BarPlot's fillStyle)
  * #977 (HypothesisTest::ChiSquared is bogus)
  * #982 (LinearModelTest::LinearModelDurbinWatson with several factors)
+ * #980 (CorrelationAnalysis_SRC scales the coefficients so that they sum to 1)
```
Comment (jschueller) [inline ChangeLog]: could you reorder these ?

**Comment by sofianehaddad** [comment]
@jschueller done

File: lib/src/Base/Stat/CorrelationAnalysis.cxx [hunk 4]
```diff
@@ -59,7 +59,8 @@ Point CorrelationAnalysis::SpearmanCorrelation(const Sample & inputSample,
 
 /* Compute the Standard Regression Coefficients (SRC) between the input sample and the output sample */
 Point CorrelationAnalysis::SRC(const Sample & inputSample,
-                               const Sample & outputSample)
+                               const Sample & outputSample,
+                               const Bool scale)
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Base/Stat/CorrelationAnalysis.cxx]: @sofianehaddad if you could address regis' last remark, i'd be happy to merge that pr

[the remark this answers is not in the document]
Comment (sofianehaddad) [reply lib/src/Base/Stat/CorrelationAnalysis.cxx]: done

**Comment by mbaudin47** [comment]
Thank you very much for the improvement. 

In the CorrelationAnalysis_SRC function, the important point is that the default value of the normalizebool argument is false. This produces unnormalized SRC coefficients, which may reveal an inappropriate linear model. 

See the following example with OT 1.12.
```
import openturns as ot
g = ot.SymbolicFunction(["X1","X2"],["X1*X2"])
X1 = ot.Normal(0.,10.)
X2 = ot.Uniform(-1.,1.)
X = ot.ComposedDistribution([X1,X2])
inputDescription = ["X1","X2"]
X.setDescription(Xdescr)
n = 10000
sampleX = X.getSample(n)
sampleY = g(sampleX)
SRCindices = ot.CorrelationAnalysis.SRC(sampleX, sampleY)
ot.SaltelliSensitivityAlgorithm.DrawImportanceFactors(SRCindices, inputDescription, 'SRC coefficients')
```

This produces the following **wrong** figure.

index

The following script fixes the bug.

```
SRC1 = ot.CorrelationAnalysis_PearsonCorrelation(sampleY,sampleX[:,0])[0]
SRC2 = ot.CorrelationAnalysis_PearsonCorrelation(sampleY,sampleX[:,1])[0]
SRC1 = max(SRC1,0.)
SRC2 = max(SRC2,0.)
unexplained = 1-SRC1-SRC2
ot.Pie([SRC1,SRC2,unexplained],["X1","X2","Unexplained"])
```

This produces the following figure.

index

In this particular case, the SRC indices are equal to zero: the interactions are major.

With OT 1.13 the new function DrawCorrelationCoefficients will allow to get the correct graphics (avoiding the infamous Pie chart), with the correct values.

**Comment by sofianehaddad** [comment]
@mbaudin47 indeed. But I think Pie drawables should not be used with this kind of feature importances. Have a look at #900

**Pull request merged**